File: flutter_tictactoe/lib/ResultScreen.dart
import 'package:flutter/material.dart';
import 'ui/theme/typography.dart';

class ResultScreen extends StatelessWidget {
  final String resultMessage;
  final VoidCallback onPlayAgain;
  final VoidCallback onBackToSettings;

  ResultScreen({
    required this.resultMessage,
    required this.onPlayAgain,
    required this.onBackToSettings,
  });

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Theme.of(context).colorScheme.background,
      body: Padding(
        padding: const EdgeInsets.all(16.0),
        child: Center(
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            crossAxisAlignment: CrossAxisAlignment.center,
            children: [
              Text(
                resultMessage,
                style: AppTypography.headlineLarge.copyWith(color: Theme.of(context).colorScheme.onBackground),
                textAlign: TextAlign.center,
              ),
              SizedBox(height: 32),
              ElevatedButton(
                onPressed: onPlayAgain,
                style: ElevatedButton.styleFrom(
                  backgroundColor: Theme.of(context).colorScheme.primary,
                  minimumSize: Size(150, 50),
                  padding: EdgeInsets.symmetric(vertical: 8.0),
                ),
                child: Text(
                  "Play again",
                  style: AppTypography.bodyMedium.copyWith(color: Colors.black)
                ),
              ),
              SizedBox(height: 32),
              ElevatedButton(
                onPressed: onBackToSettings,
                style: ElevatedButton.styleFrom(
                  backgroundColor: Theme.of(context).colorScheme.primary,
                  minimumSize: Size(150, 50),
                  padding: EdgeInsets.symmetric(vertical: 8.0),
                ),
                child: Text(
                  "Back to settings",
                    style: AppTypography.bodyMedium.copyWith(color: Colors.black)
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}

File: Kotlin/app/src/main/java/com/example/tictactoe/ResultScreen.kt
package com.example.tictactoe

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp

@Composable
fun ResultScreen(
    resultMessage: String,
    onPlayAgain: () -> Unit,
    onBackToSettings: () -> Unit
) {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(MaterialTheme.colorScheme.background)
            .padding(16.dp),
        verticalArrangement = Arrangement.Center,
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        Text(
            text = resultMessage,
            style = MaterialTheme.typography.headlineLarge,
            color = MaterialTheme.colorScheme.onBackground,
            modifier = Modifier.padding(bottom = 32.dp)
        )

        Button(
            onClick = onPlayAgain,
            modifier = Modifier
                .padding(16.dp)
                .size(150.dp, 50.dp),
            colors = ButtonDefaults.buttonColors(
                contentColor = Color.Black)
        ) {
            Text("Play again")
        }

        Button(
            onClick = onBackToSettings,
            modifier = Modifier
                .padding(16.dp)
                .size(150.dp, 50.dp),
            colors = ButtonDefaults.buttonColors(
                contentColor = Color.Black)
        ) {
            Text("Back to settings")
        }
    }
}
